File: snippets/python/20-search/03-fulltext.py
# import the shared client:  from elo_playground import connect
# topic:    Full-text search
# category: Search
# id:       search.fulltext

import os
from elo_playground import connect

# --- local ELO test box (override with ELOPG_* env vars or a .env) ---
ELO_BASE_URL = os.getenv("ELOPG_ELO_BASE_URL", "http://localhost:9090/ix-Repository1") # ELOPG_DEFAULT:base_url
ELO_USER = os.getenv("ELOPG_ELO_USER", "Administrator") # ELOPG_DEFAULT:user
ELO_PASS = os.getenv("ELOPG_ELO_PASSWORD", "elo") # ELOPG_DEFAULT:password

elo = connect(base_url=ELO_BASE_URL, user=ELO_USER, password=ELO_PASS)

rows = elo.find_all(
    "findFirstSords", "findNextSords", "sords",
    {
        "findInfo": {"findByFulltext": {"fulltext": "agreement", "isTree": False}},
        "max": 50,
        "sordZ": {"bset": "449304431574384639"},
    },
)

print(f"{len(rows)} documents match 'agreement'")
for s in rows:
    print(f"  {s['name']}")
if not rows:
    print("  (0 hits usually means the ELO full-text service is not running "
          "or has not finished indexing on this instance)")

File: snippets/node/20-search/03-fulltext.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    Full-text search
// category: Search
// id:       search.fulltext

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_USER = "Administrator";
const ELO_PASS = "elo";

const elo = await connect({ user: ELO_USER, password: ELO_PASS });

const rows = await elo.findAll("findFirstSords", "findNextSords", "sords", {
  findInfo: { findByFulltext: { fulltext: "agreement", isTree: false } },
  max: 50,
  sordZ: { bset: "449304431574384639" },
});

console.log(`${rows.length} documents match 'agreement'`);
rows.forEach((s) => console.log(`  ${s.name}`));
if (!rows.length) {
  console.log("  (0 hits usually means the ELO full-text service is not running " +
    "or has not finished indexing on this instance)");
}
